### DISCUSSION TITLE
libpag.wasm:0x1a07b8 Uncaught (in promise) RuntimeError: memory access out of bounds

### DISCUSSION THREAD

**Discussion by leohandsome1688** [Tencent/libpag#2033] [Q&A]
![image](https://github.com/Tencent/libpag/assets/153789735/39e39dd6-ae7f-42ab-b643-ace0b7acf92c)
报错复现的原因：我在页面里面渲染一个pag文件，在这个页面里面，我有一个弹窗组件，里面是上传的pag文件列表，我重复打开这个弹窗选要编辑的pag文件，就会报这个错误

**Comment by jinphic** [comment]
> ![image](https://private-user-images.githubusercontent.com/153789735/292423386-39e39dd6-ae7f-42ab-b643-ace0b7acf92c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUxODY0MzgsIm5iZiI6MTc2NTE4NjEzOCwicGF0aCI6Ii8xNTM3ODk3MzUvMjkyNDIzMzg2LTM5ZTM5ZGQ2LWFlN2YtNDJhYi1iNjQzLWFjZTBiN2FjZjkyYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwOFQwOTI4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZDcxZWNjYjEyMDRkM2QxMmUyMWQ0MjY2OTI4ZjI3MDc1ZDhmMWExZmZkNzA4YmViZWRiNDQ0ZGQxOGJiZjg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7x_jqqfelLdfGe5qYAlZbjs416XLiZtRvgVo_SA5ZPg) 报错复现的原因：我在页面里面渲染一个pag文件，在这个页面里面，我有一个弹窗组件，里面是上传的pag文件列表，我重复打开这个弹窗选要编辑的pag文件，就会报这个错误

我也遇到了，有没有老哥有经验的，这个表现是怎么样的

**Comment by jinphic** [comment]
const handleUnhandledRejection = (event: PromiseRejectionEvent) => {
      if (event?.reason && typeof event?.reason === 'object') {
        const sourceURL = event?.reason.sourceURL || (event?.reason.stack && event?.reason.stack.includes('libpag'));
        if (sourceURL && sourceURL.includes('libpag')) {
          event?.preventDefault(); // 阻止浏览器默认错误处理
          // do something backup 兜底错误处理
          return;
        }
      }
    };
    window.addEventListener('unhandledrejection', handleUnhandledRejection);
    
    我加了这个处理好像没有捕捉到， 外网还是有这种错误日志，这个是错误不是 promisereject 吗..

**Reply by jinwuwu001** [reply]
请描述下使用的 libpag 版本、平台等信息，最好再提供一个可复现问题的 demo.

**Reply by jinphic** [reply]
我这边没办法复现，外网用户使用实际过程中上报的错误信息

**Comment by jinphic** [comment]
> ![image](https://private-user-images.githubusercontent.com/153789735/292423386-39e39dd6-ae7f-42ab-b643-ace0b7acf92c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUzMzczNjMsIm5iZiI6MTc2NTMzNzA2MywicGF0aCI6Ii8xNTM3ODk3MzUvMjkyNDIzMzg2LTM5ZTM5ZGQ2LWFlN2YtNDJhYi1iNjQzLWFjZTBiN2FjZjkyYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxMFQwMzI0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZjg3M2RjYmUzNjI5OGI1MjliMWM5OGRiZmFhNGFkNTY1NTFjNzgwZjY1ZmY2MGUzYzkxYzNhYzNhZjRkNzY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EDyZM2DgF0g5foginc9YvK8bmbOCmPizwqlI5FJT-6Q) 报错复现的原因：我在页面里面渲染一个pag文件，在这个页面里面，我有一个弹窗组件，里面是上传的pag文件列表，我重复打开这个弹窗选要编辑的pag文件，就会报这个错误

hello，请问你这边有可运行的demo吗，我想尝试下这样能不能捕获到错误内容。 我本地没办法复现，也不知道外网用户怎么触发的
```
    window.addEventListener('error', handleError);
    const handleError = (event: ErrorEvent) => {
      if (event?.error && typeof event?.error === 'object') {
        const sourceURL = event?.error.sourceURL || (event?.error.stack && event?.error.stack.includes('libpag'));
        if (sourceURL && sourceURL.includes('libpag')) {
          event?.preventDefault(); // 阻止浏览器默认错误处理
          // do something
        }
      }
    };
```
